components/prism-javascript.js

```diff
@@ -29,26 +29,26 @@ Prism.languages.insertBefore('javascript', 'keyword', {
 	},
 	// This must be declared before keyword because we use "function" inside the look-forward
 	'function-variable': {
-		pattern: /[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*(?=\s*[=:]\s*(?:async\s*)?(?:\bfunction\b|(?:\([^()]*\)|[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*)\s*=>))/i,
+		pattern: /[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*(?=\s*[=:]\s*(?:async\s*)?(?:\bfunction\b|(?:\(.*\)|[_$a-z\xA0-\uFFFF][$\w\xA0-\uFFFF]*)\s*=>))/i,
```

Comment:
You're right... I want to match recursive brackets, but it's not the thing that pure regex can do, so I thought matching the start `(` and the end `)` is enough.

The finite recursive depth trick is fine in most cases, I'll take your advice.

We can have neither strings nor comments in parameters because they'll split the codes before `function-variable` and `parameter` get to work, this is another example that comments break the highlight in Prism (what a pity):

```jsx
<Button
    prop1={value1}
    // some comments
/>
```

Reply:
Yeah, comments and strings are a huge pain...